examples/python/ros/main.py

```diff
@@ -0,0 +1,275 @@
+        self.cv_bridge = cv_bridge.CvBridge()
+        self.laser_proj = laser_geometry.laser_geometry.LaserProjection()
+
+        # Log a bounding box as a visual placeholder for the map
+        rr.log_obb(
+            "map/box",
+            half_size=[6, 6, 2],
+            position=[0, 0, 1],
+            color=[255, 255, 255, 255],
+            timeless=True,
+        )
+
+        # Subscriptions
+        self.info_sub = self.create_subscription(
+            CameraInfo,
+            "/intel_realsense_r200_depth/camera_info",
+            self.cam_info_callback,
+            10,
+            callback_group=self.callback_group,
+        )
+
+        self.odom_sub = self.create_subscription(
+            Odometry,
+            "/odom",
+            self.odom_callback,
+            10,
+            callback_group=self.callback_group,
+        )
+
+        self.img_sub = self.create_subscription(
+            Image,
+            "/intel_realsense_r200_depth/image_raw",
+            self.image_callback,
+            10,
+            callback_group=self.callback_group,
+        )
+
+        self.points_sub = self.create_subscription(
+            PointCloud2,
+            "/intel_realsense_r200_depth/points",
+            self.points_callback,
+            10,
+            callback_group=self.callback_group,
+        )
+
+        self.scan_sub = self.create_subscription(
+            LaserScan,
+            "/scan",
+            self.scan_callback,
+            10,
+            callback_group=self.callback_group,
+        )
+
+        # The urdf is published as latching
+        self.urdf_sub = self.create_subscription(
+            String,
+            "/robot_description",
+            self.urdf_callback,
+            qos_profile=latching_qos,
+            callback_group=self.callback_group,
+        )
+
+    def log_tf_as_rigid3(self, path: str, time: Time) -> None:
+        """Helper to look up a transform with tf and log using `log_rigid3`."""
+        # Get the parent path
+        parent_path = path.rsplit("/", 1)[0]
+
+        # Find the corresponding frames from the mapping
+        child_frame = self.path_to_frame[path]
+        parent_frame = self.path_to_frame[parent_path]
+
+        # Do the TF lookup to get transform from child (source) -> parent (target)
+        try:
+            tf = self.tf_buffer.lookup_transform(parent_frame, child_frame, time, timeout=Duration(seconds=0.1))
+            t = tf.transform.translation
+            q = tf.transform.rotation
+            rr.log_rigid3(path, parent_from_child=([t.x, t.y, t.z], [q.x, q.y, q.z, q.w]))
+        except TransformException as ex:
+            print("Failed to get transform: {}".format(ex))
+
+    def cam_info_callback(self, info: CameraInfo) -> None:
+        """Log a `CameraInfo` with `log_pinhole`."""
+        time = Time.from_msg(info.header.stamp)
+        rr.set_time_nanos("ros_time", time.nanoseconds)
+
+        self.model.fromCameraInfo(info)
```

Comment:
It's just how the helper library happens to be implemented. Mostly I think it updates state because then that object is then used for subsequent projection operations (which we don't need).